matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/auth/registration/RegisterTask.kt

```diff
@@ -40,8 +41,20 @@ internal class DefaultRegisterTask(
             }
         } catch (throwable: Throwable) {
             throw throwable.toRegistrationFlowResponse()
-                    ?.let { Failure.RegistrationFlowError(it) }
-                    ?: throwable
+                ?.let { Failure.RegistrationFlowError(it) }
+                ?: throwable
         }
     }
 }
+
+internal class RegisterAvailableTask(private val authAPI: AuthAPI) : Task<String, Availability> {
```

Comment:
Please move this to its own file and create an interface for this new task (so how it is done for the other Task)

Reply:
Partially done in done in https://github.com/vector-im/element-android/pull/3084